frontend/src/reviews/ReviewComments.tsx

```diff
@@ -31,23 +45,107 @@ export default function ReviewComments({ release, accessRequest }: ReviewComment
     ...semverOrAccessRequestIdObject,
   })
 
-  const reviewDetails = useMemo(
-    () =>
-      reviews.map((review) =>
-        review.responses.map((response) => <>{response.decision && <ReviewDecision response={response} />}</>),
-      ),
-    [reviews],
-  )
+  const reviewDetails = useMemo(() => {
+    let decisionsAndComments: Array<ReviewResponseKind> = []
+    reviews.forEach((review) => {
+      review.responses.forEach((response) => decisionsAndComments.push(response))
+    })
+    if (release && release.comments) {
+      decisionsAndComments = [...decisionsAndComments, ...release.comments]
+    }
+    if (accessRequest && accessRequest.comments) {
+      decisionsAndComments = [...decisionsAndComments, ...accessRequest.comments]
+    }
+    decisionsAndComments.sort(sortByCreatedAtAscending)
+    return decisionsAndComments.map((response) => {
+      if (isReviewResponse(response)) {
+        return <ReviewDecisionDisplay key={response.createdAt} response={response as ReviewResponse} />
+      } else {
+        return <ReviewCommentDisplay key={response.createdAt} response={response as ReviewComment} />
+      }
+    })
+  }, [reviews, release, accessRequest])
+
+  async function submitReviewComment() {
+    setCommentSubmissionError('')
+    if (!newReviewComment) {
+      setCommentSubmissionError('Please provide a comment before submitting.')
+      return
+    }
+    if (currentUser) {
+      const newComment: ReviewComment = {
+        comment: newReviewComment,
+        user: currentUser.dn,
+        createdAt: new Date().toISOString(),
+      }
+      if (release) {
+        const updatedRelease: ReleaseInterface = release
+        if (!updatedRelease.comments) {
```

Comment:
This logic is strange, shouldn't there always be a `release.comments` array? We can create a migration script to add the array to any existing data and should be able to guarantee that any new releases are created with an empty comments array.